website/docs/core-concepts/stacks/yaml-functions/store.mdx

```diff
@@ -0,0 +1,153 @@
+  !store <component> {{ .atmos_stack }} <key>
+```
+
+For example, you have a `tgw` component that requires the `vpc_id` key from the store for the `vpc` component in the same stack:
+
+```yaml
+  components:
+    terraform:
+      tgw:
+        vars:
+          vpc_id: !store vpc {{ .stack }} vpc_id
+```
+
+:::note Using the `.stack` or `.atmos_stack` template identifiers to specify the stack is the same as calling the
+`!store` function with two parameters without specifying the current stack, but without using `Go` templates. If you
+need to get an value from the store froma component in the current stack, using the `!store` function with two
+parameters is preferred because it has a simpler syntax and executes faster. :::
+
+### Use a Format Function
+
+Use the `printf` template function to construct stack names using static strings and dynamic identifiers.
+This is convenient when you want to override some identifiers in the stack name:
+
+```yaml
+  !store <component> {{ printf "%s-%s-%s" .vars.tenant .vars.environment .vars.stage }} <key>
+
+  !store <component> {{ printf "plat-%s-prod" .vars.environment }} <key>
+
+  !store <component> {{ printf "%s-%s-%s" .settings.context.tenant .settings.context.region .settings.context.account }} <key>
+```
+
+<dl>
+  <dt>`<component`</dt>
+  <dd>Placeholder for an actual component name (e.g. `vpc`)</dd>
+  <dt>`<key>`</dt>
+  <dd>Placeholder for an actual key (e.g. `subnet_ids`)</dd>
+</dl>
+
+
+For example, you have a `tgw` component deployed in the stack `plat-ue2-dev`. The `tgw` component requires the
+`vpc_id` key from the store for the `vpc` component from the same environment (`ue2`) and same stage (`dev`), but from a different
+tenant `net` (instead of `plat`):
+
+```yaml title="plat-ue2-dev"
+  components:
+    terraform:
+      tgw:
+        vars:
+          vpc_id: !store vpc {{ printf "net-%s-%s" .vars.environment .vars.stage }} vpc_id
+```
+
+:::tip Important
+    By using the `printf "%s-%s-%s"` function, you are constructing stack names using the stack context variables/identifiers.
+
+    For more information on Atmos stack names and how to define them, refer to `stacks.name_pattern` and `stacks.name_template`
+    sections in [`atmos.yaml` CLI config file](/cli/configuration/#stacks)
+:::
+
+## Considerations
+
+ - Using `!store` with secrets can expose sensitive data to standard output (stdout) in any commands that describe stacks or components.
+- When using `!store` with [`atmos describe affected`](/cli/commands/describe/affected), Atmos requires access to all referenced remote states.
+   If you operate with limited permissions (e.g., scoped to `dev`) and reference production stacks, the command will fail.
+- Be mindful of disaster recovery (DR) implications when using it across regions.
+- Consider cold-start scenarios: if the dependent component has not yet been provisioned, the value in the store may not yet be available and the `!store` function call will fail.
```

Comment:
Do we provide a workaround with defaults?

Reply:
Not yet. I will add that in a follow-up PR.